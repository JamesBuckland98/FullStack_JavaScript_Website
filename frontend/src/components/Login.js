import '../css/Login.css'
import login from '../api/loginApi';
import React, { Component } from 'react';
import { Formik, Form, Field, ErrorMessage } from 'formik';

class Login extends Component {

	constructor(props) {
		super(props)
		this.state = {
			responseText: '',
		}
	}
	render() {
		return (
			<div>
				<p> { this.state.responseText } </p>
				<h3> Login: </h3>
				<Formik
					initialValues={{ username : '', password: '' }}
					validate= {
						values => {
							let errors = {};
							if (!values.username) {
								errors.username = 'Required';
							}
							if (!values.password) {
								errors.password = 'Required';
							}
							return errors;
						}
					}
					onSubmit={(values, { setSubmitting }) => {
						setTimeout(() => {
							login(values, (err, data) => {
								if (data instanceof Object) {
									this.props.setUsername(data.Username)
									this.props.switch('login')
									this.props.switch('chat')
								} else {
									this.setState({
										responseText: 'Incorrect username or password'
									})
								}
							});
							setSubmitting(false);
						}, 400);
					}}
				>
					{({ isSubmitting }) => (
						<Form>
							<label> Username: </label> <br/>
							<Field type='text' name='username' />
							<ErrorMessage name='username' component='div' />
							<br/>
							<label> Password: </label> <br/>
							<Field type='password' name='password' />
							<ErrorMessage name='password' component='div' />
							<br/>
							<button type='submit' disabled={isSubmitting}>
								Submit
							</button>
						</Form>
					)}
				</Formik>
				<button className='link' onClick= {() => {
					this.props.switch('login'); 
					this.props.switch('sign up');
				}}>
					Sign Up
				</button>
			</div>
		)
	}
}

export default Login;