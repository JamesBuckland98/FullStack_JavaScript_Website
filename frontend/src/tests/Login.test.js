import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from '../components/Login';


it('should render Login component correctly', () => {
	const wrapper = shallow(<Login/>);
	expect(toJson(wrapper)).toMatchSnapshot();
})


describe('When sign up button is clicked', () => {
	it('switches between login and sign up', () => {
		const switchComponet = jest.fn();
		const wrapper = shallow(<Login switch= { switchComponet }/>);
		const buttonElement = wrapper.find('.link')
		buttonElement.simulate('click');
		expect(switchComponet).toHaveBeenCalledTimes(2);
	})
})