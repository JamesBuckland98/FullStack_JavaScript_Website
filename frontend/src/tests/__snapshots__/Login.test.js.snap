// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component correctly 1`] = `
<div>
  <p>
     
     
  </p>
  <h3>
     Login: 
  </h3>
  <Formik
    initialValues={
      Object {
        "password": "",
        "username": "",
      }
    }
    onSubmit={[Function]}
    validate={[Function]}
  >
    <Component />
  </Formik>
  <button
    className="link"
    onClick={[Function]}
  >
    Sign Up
  </button>
</div>
`;
